src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -527,9 +539,9 @@ public void setQuestionDetails(FeedbackQuestionDetails questionDetails) {
      * @return The Feedback*QuestionDetails object representing the question's details
      */
     public FeedbackQuestionDetails getQuestionDetails() {
-        // For Text questions, the questionText simply contains the question, not a JSON
+        // For old Text questions, the questionText simply contains the question, not a JSON
         // This is due to legacy data in the data store before there are multiple question types
-        if (questionType == FeedbackQuestionType.TEXT) {
+        if (questionType == FeedbackQuestionType.TEXT && !isJsonValid(questionMetaData.getValue())) {
             return new FeedbackTextQuestionDetails(questionMetaData.getValue());
```

Comment:
this is to handle previous Questions i.e that contain question as Text not as Json string.